src/index.js

```diff
@@ -1,31 +1,20 @@
 const borc = require('borc')
 const { newFromString, encode } = require('@openworklabs/filecoin-address')
+const BigNumber = require('bignumber.js')
 const { marshalBigInt } = require('./utils')
 
-class Message {
-  constructor({ to, from, nonce, value, method, gasPrice, gasLimit, params }) {
-    if (!nonce) throw new Error('No nonce provided or nonce is not a string')
-    this.nonce = nonce
+let typeCheck
```

Comment:
since the value of this isn't changing - lets just define `typeCheck` as a `const` at the bottom.

`const typeCheck = ({ to, from... }) => {...}`

Reply:
Word, I kinda like the style where you define what you have to work with up top, but in this case I don't mind moving it down.

Reply:
yeah, plus it leaves less up to the runtime interpreter, which doesn't always act the way we'd expect.